### ISSUE TITLE
Question about how should a mutation be implemented using `graphqlUploadExpress`

### ISSUE THREAD

**Issue by tanakaworld** [flash-oss/graphql-upload-minimal#40]
Following the example described in [the README](https://github.com/flash-oss/graphql-upload-minimal/blob/cdc130891f8892089f661cbbfc0ab1958dc3b23b/README.md#expressjs), it is likely fields defined as `Upload` in the schema would be an array of `Promise`. Is it expected? 

If that's correct, then the example should be like the following:

```ts
// As Is:
const { createReadStream, filename /*, fieldName, mimetype, encoding */ } = await doc.file;

// To Be:
const { createReadStream, filename /*, fieldName, mimetype, encoding */ } = (await doc).file;
```

The environment I tested:

- graphql: `16.6.0`
- graphql-upload-minimal: `1.5.4`
- @apollo/server: `4.3.2`
- express: `4.18.2`

**Comment by koresar** [comment]
I do not have an answer for "is it expected". I didn't get the question.

But I'm sure this piece of README is very much correct.

**Comment by tanakaworld** [comment]
I mean `doc.file` is not a promise based on [the type definition](https://github.com/flash-oss/graphql-upload-minimal/blob/cdc130891f8892089f661cbbfc0ab1958dc3b23b/public/index.d.ts#L39). I suspect `doc` is a promise object do the example should be `(await doc).file;`

**Comment by koresar** [comment]
You are right, the `doc.file` does not look like a Promise object in the TypeScript. Although, it's a Promise object, I've just double checked. Perhaps, you can fix that with a Pull Request?

The `doc` definitely is not a promise. It's a regular object.

**Comment by tanakaworld** [comment]
Thanks for the double-check. Sorry, I realized that I had mistakenly understood the type definition—this [type definition](https://github.com/flash-oss/graphql-upload-minimal/blob/cdc130891f8892089f661cbbfc0ab1958dc3b23b/public/index.d.ts#L39) is the type of the `Upload` class which did not correspond with the value available in a resolver.

So, a schema defined with the `Upload` scalar should be a promise of `FileUpload` in a resolver as I investigated https://github.com/tanakaworld/graphql-upload-sample/issues/1, meaning `DocumentUploadInput.file` in [README](https://github.com/flash-oss/graphql-upload-minimal/blob/cdc130891f8892089f661cbbfc0ab1958dc3b23b/README.md#expressjs) is a promise and `await doc.file` is correct.
